### DISCUSSION TITLE
Required checks that due configuration ignore committed changes are stuck in "Waiting for status to be reported"

### DISCUSSION THREAD

**Discussion by minherz** [community/community#49124] [Pull Requests] [answered]
### Select Topic Area

Bug

### Body

When a repo has workflows that use `paths-ignore` or another configuration that may result in not running the workflow on particular commits and this workflow is marked as a required check, it results in a PR got stuck in the state when these checks are marked as _Expected — Waiting for status to be reported_ and never complete. To merge such PR, a user has to manually bypass the branch protection.
The expected behavior would be for such checks to pass momentarily.

**Comment by minherz** [comment, marked as answer]
It looks like there is an [official solution](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks) for this case.
In the essence it is not a solution but a workaround. If the check is skipped because of [path filtering](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore), [branch filtering](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore) or a [commit message](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs), the way to complete this check is to define _another workflow_ that **does nothing** for actions that do not match conditions in your workflow. For example, for
```yaml
on:
  pull_request:
    paths:
      - 'scripts/**'
```
the complement workflow will have:
```yaml
on:
  pull_request:
    paths-ignore:
      - 'scripts/**'
```

**Reply by jayaddison** [reply]
@minherz a note to let you know that the page you linked to now seems to redirect to another page, and that redirected page does not include the solution.

I am able to find the material I think you referred to, when you were preparing this accepted answer, currently at: https://web.archive.org/web/20230321195500/https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks

**Reply by minherz** [reply]
It is not surprising since Github evolves and my previous comment was made in 2023.

**Comment by shirady** [comment]
Hi @minherz,
thank you for sharing the official documentation, but I still don't understand what changes should I do when I'm using `paths-ignore` in a required check., for example: 
```
on:
  push:
    paths-ignore:
      - 'docs/**'
   ```
Edit: taken from [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths)
How should the suggested generic workflow look like?

**Comment by minherz** [comment]
@shirady, the syntax in your code example is invalid. And I am afraid that I do not understand your question.
The current design ignores checks (jobs within a workflow) that do not match filters (e.g. branch name or paths or file extensions). It becomes a problem when the check is defined as a required. The ignored status does not satisfy the requirement for this check to be successfully executed a.k.a. to pass. The Github [documentation] describes a workaround: defining a "complimentary" workflow that has the same name both for the workflow and for the required check(s) / job(s). The documentation claims that it will resolve the problem when Push requests get stuck on waiting for required checks.

The problem is that this workaround does not work well. And, in my opinion, it will be more natural to address it via design instead of forcing users to create redundant checks that result in confusing output (same checks appear twice in many PRs).

[documentation]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

**Comment by shirady** [comment]
@minherz I fixed the code (missed a line).
Anyway, I wouldn't like to have redundant checks as you describe.
I agree with you, I don't like the workaround and I would like it to just work...

**Reply by minherz** [reply]
Thank you for feedback. Unfortunately, I could not figure out the right way to request for this feature. My support request was rejected ([ticket 2052209](https://support.github.com/ticket/2052209)).

**Reply by shirady** [reply]
I don't have access to the ticket ("Ticket not found. The ticket either does not exist or you no longer have access to it.").
Anyway, since they published the workaround, I'm sure they are aware of the need (and you published the bugs here), and I can assume it is probably not in high priority.
